.github/workflows/ant.yaml

```diff
@@ -44,19 +44,52 @@ jobs:
       - run: yarn aos:build
       - run: yarn test
 
-  notify:
+  publish:
     runs-on: ubuntu-latest
+    environment: main
     needs: [unit, integration]
+    permissions:
+      contents: write # to be able to publish a GitHub release
+    outputs:
+      published_lua_code_id: ${{ steps.publish-lua-code.outputs.srcTxId }}
+      # lua_code: ${{ steps.publish-lua-code.outputs.luaSourceCode }}
+    if: github.ref == 'refs/heads/main'
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4.0.2
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'yarn'
+      - run: yarn --frozen-lockfile
+      - name: Publish Lua Code
+        id: publish-lua-code
+        env:
+          WALLET: ${{ secrets.WALLET }}
+          GITHUB_SHA: ${{ github.sha }}
+        run: |
+          PUBLISH_OUTPUT=$(yarn aos:publish 2>&1)
+          LUA_SOURCE_CODE_TX_ID=$(echo "$PUBLISH_OUTPUT" | grep -oE "publishedLuaTxId\.[a-zA-Z0-9_-]{43}" | cut -d'.' -f2)
+          echo "::set-output name=srcTxId::$LUA_SOURCE_CODE_TX_ID"
+
+      - name: Create Tag for Lua ID
+        run: |
+          git tag "${{ steps.publish-lua-code.outputs.srcTxId }}"
+          git push --tags
+
+  notify:
+    runs-on: ubuntu-latest
+    needs: [unit, integration, publish]
     if: always()
     steps:
       - name: Notify Slack on Success
-        if: success() && github.ref == 'refs/heads/main' && needs.unit.result == 'success' && needs.integration.result == 'success'
+        if: github.ref == 'refs/heads/main' && needs.unit.result == 'success' && needs.integration.result == 'success' && (needs.publish.result == 'success' || needs.publish.result == 'skipped')
         uses: 8398a7/action-slack@v3
         with:
           status: ${{ job.status }}
           text: |
             Unit Tests: ${{ needs.unit.result }}
             Integration Tests: ${{ needs.integration.result }}
+            Published Lua Id: ${{ needs.publish.outputs.published_lua_code_id}}
```

Comment:
this could be null - will this handle correctly?

Reply:
Not sure - added a case for null to show 'N/A' if the lua id isnt there.